keps/498-GroupScheduling/README.md

```diff
@@ -0,0 +1,392 @@
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+
+## Motivation
+
+Many users of managed K8s services on cloud providers make use of NAP (node auto provisioning) which creates node pools for pending/unschedulable pods, based on those pods requirements (i.e., CPU/memory requirements, GPU/TPU requirements, etc).
+
+Since node pool provisioning takes a variable amount of time, users are running into issues where the first slice finishes provisioning and pods land there and begin running, but eventually timeout before the other slices all finish provisioning and pods land there and become ready.
+
+For statically provisioned clusters, we have recommended users use Kueue to handle group scheduling of JobSets once sufficient resources are available.
+
+However, for dynmaically provisioned clusters, we have no support for group scheduling in JobSet yet.
+
+### Goals
+
+- Enhance the JobSet API with options allowing the user to define a group scheduling policy.
+
+### Non-Goals
+
+- Upstream changes that would improve/simplify group scheduling support in JobSet.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1: Group scheduling of all replicated jobs
+
+As a user, in order to make efficient use of expensive accelerator (GPU/TPU) resources, I use node
+auto-provisioning to provision infrastructure on an as-needed basis when a pending workload requires
+it, then deprovision it once it's no longer in use. Since the completion time of these provisioning
+operations is variable, I want to make sure JobSet pods which land on the earliest provisioned nodes
+do not start executing the main container until all the infrastructure is provisioned and
+all pods have started. Otherwise, the distributed initialization step will timeout in the earliest
+pods while waiting for the remaining provisioning to complete.
+
+
+### Example JobSet spec using group scheduling for all replicated jobs:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: my-jobset
+  namespace: default
+spec:
+  # main containers in pods part of all replicatedJob will be blocked
+  # from execution until all pods have started, or timeout after 5min.
+  groupSchedulingConfig:
+    timeoutAfterSeconds: 300
+  replicatedJobs:
+    - name: workers
+      replicas: 3
+      template:
+        spec:
+          backoffLimit: 0
+          completions: 100
+          parallelism: 100
+          template:
+            spec:
+              containers:
+                - name: pytorch
+                  image: pytorch/pytorch:latest
+                  command:
+                    - bash
```

Comment:
its here so I'll comment on it..

I don't think bash will run a python script. :)

Reply:
Woops! Fixed